backup/backup.go

```diff
@@ -373,6 +373,8 @@ func Backup(pctx context.Context, jobInfo *files.JobInfo) error {
 				case fileBuffer <- true:
 				}
 			case <-ctx.Done():
+				log.AppLogger.Debugf("manifest copy: ctx.Done(): err = %v", ctx.Err())
```

Comment:
**concern:** I'm not sure this is the correct spot for this - line 287 should trigger above should any error raise from the underlying zfs backup stream, perhaps actually we should add this to line 300 while waiting to pass along the volume? 

Also looking through here, line 358 may also warrant a call to `maniwg.Done()`...

Perhaps a larger issue here is all the corner cases this method of sync'ing between goroutines has and addressing the TODO below on line 385.